woocommerce/woocommerce-ios
# [Mobile Payments] Show card's brand and last 4 digits on refunds

Closes: #5976


### Description

Adds the Card Brand and Last4 digits to the Refund Confirmation screen for Card Present Payment transactions. This is to help the customer/merchant perform refunds onto the same card which was used for payment.

### Remaining work on 5976
- Test that we fetch the charge when we reach the refund screen, not just display it correctly when passed in #6180
- Do we want to display this for `card` transactions, not just `cardPresent`? I can see arguments both ways but it's straightforward to do. **Leaving this for now, we can add it later if we want to.**
- Stripe gateway support #6181

### Testing instructions

To test, a WooCommerce Payments In Person Payment setup is required. See PdfdoF-D-p2 for info.

1. In the Orders tab, tap an order which has been paid for by a card-present payment
2. Tap `Refund`
3. Select a product to refund or toggle `Refund fees` if required, then tap `Next`
4. Observe that the `Refund Via` row contains card brand and last 4 digits

You can use the [testing card numbers that Stripe provide](https://stripe.com/docs/terminal/references/testing#standard-test-cards) to test different brands/last4 combos with the simulated reader.

N.B. If you follow these instructions and refund immediately after taking payment `Refund Via` screen will not always contain the card details, instead it may just say `WooCommerce Payments`. This is a known issue to be addressed in a future PR.

### Screenshots


Visa | Diners Club
---|---
![Visa card present refund](https://user-images.githubusercontent.com/2472348/153452013-090bbfc1-5724-490f-ba3e-c952369828cc.png) | ![Diners card present refund](https://user-images.githubusercontent.com/2472348/153452018-68bf7777-0a91-4bdc-a2db-e4b06f8dc9b4.png)


https://user-images.githubusercontent.com/2472348/153456499-9a8f0407-a256-4292-854f-32f3ff34a7b5.mp4

@woocommerce/mobile-designers I've gone for two rows here, because card brands can be a little long so combining them with `WooCommerce Payments` on the same line as in the design would lead to them being truncated. Do you think we should reverse the order of the strings here, so that the card details are the title and a little more prominent?

---
